### PR TITLE
v1.2 open tests & other minor ammendments

### PR DESCRIPTION
**Pull request by Joe-Dunleavy**
- updated v1.2 open tests (from Oli)
- minor amendments to test reference tags
- removed duplicate test

### CODE REVIEW

File: tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts [hunk 1]
```diff
@@ -10,8 +10,8 @@ const docs =
   "\r\nCause: ";
 
 export default () =>
-  describe("fdc3.raiseIntentForContext", async () => {
-    it("(BasicRI1) Method is callable", async () => {
+  describe("(NoContext) fdc3.raiseIntentForContext", async () => {
```
Comment (robmoffat) [inline tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts:13]: In the sheet and in the 1.2 Test Definitions the (NoContext) test is part of the Intents section.  If we're going to change it here we also need to change the Test Definition PR in the FDC3 project.  

I suggest you keep the test where it was if possible so we don't have to do that work, even if it makes BasicRI1 a subset of NoContext.

**Review by robmoffat — CHANGES_REQUESTED** [review]

File: tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts (hunk 1 above)
Comment (robmoffat) [reply tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts:13]: https://github.com/finos/FDC3/pull/848

Comment (Joe-Dunleavy) [reply tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts:13]: Ok, I'll move it back. In that case, I think we can remove the basic raiseIntentForContext.ts as the test within is a duplicate

Comment (Joe-Dunleavy) [reply tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts]: I've added that back in where it was

File: tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts [hunk 2]
```diff
@@ -10,18 +10,12 @@ const docs =
   "\r\nCause: ";
 
 export default () =>
-  describe("(NoContext) fdc3.raiseIntentForContext", async () => {
+  describe("fdc3.raiseIntentForContext", async () => {
     it("(NoContext) Passing an invalid context causes a NoAppsFound error to be thrown", async () => {
```
Comment (robmoffat) [inline tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts:14]: this also needs changing back

**Review by robmoffat — CHANGES_REQUESTED** [review]

File: tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts (hunk 1 above)
Comment (robmoffat) [reply tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts]: nearly right - see my review

File: tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts (hunk 2 above)
Comment (Joe-Dunleavy) [reply tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts]: done

Comment (Joe-Dunleavy) [reply tests/src/test/v1.2/basic/fdc3.raiseIntentForContext.ts]: I've changed the tag name back to what it was, but the tag shouldn't have been in the `describe`  either, just in the `it`

**Review by robmoffat — APPROVED** [review]

**Comment by mhmcclung** [comment]
Hi Joe.  I think I spotted a problem that might surface in multiple places, but I focused on diagnosing just one test, AOpensBWithContext1Test.  
 
The problem is probably intermittent, dependent on message timing, which may be why it didn't surface early. 
 
I saw the test fail because there were two incoming broadcasts messages to the conformance app with context type "fdc3-conformance-context-received".  The test validated the values in the first broadcast message, but the right data was in the second broadcast message.  
 
Going into a little more detail, the first incoming broadcast message (the wrong one) has a context type of "fdc3-conformance-context-received" with a child context of type “fdc3-intent-c-opened”.  The second incoming broadcast (the right one) also has a context type of "fdc3-conformance-context-received" but with a child context of type fdc3.genericListener. 
 
The first/wrong message was sent due to the way mock\intent-c\index.html is coded.  Note that after the channel is joined, fdc3-intent-c-opened" is broadcast, then an addListener with a null context is invoked -- this addListener will automatically receive the state of the channel and trigger the extra "fdc3-conformance-context-received" broadcast.

**Comment by Joe-Dunleavy** [comment]
@mhmcclung Thanks for flagging this up Mike. Although I didn't write the open tests, I did test them before making this pull request and missed this due to the tests passing when I ran them. 

I've addressed the issue by passing `"fdc3.testReceiverMultiple"` to the listener instead of `null`.

@robmoffat here's a PR that addresses this:
https://github.com/finos/FDC3-conformance-framework/pull/114

**Pull request merged**